### PR TITLE
Fix View layout issues

### PR DESCRIPTION
**Pull request by infosia**
* [TIMOB-24883](https://jira.appcelerator.org/browse/TIMOB-24883) Incorrect child view size when parent view has border
* [TIMOB-24848](https://jira.appcelerator.org/browse/TIMOB-24848) ListView template with left and right defined causes crash
* [TIMOB-24831](https://jira.appcelerator.org/browse/TIMOB-24831) Layout does not seem to consider left and right properties
* [TIMOB-24828](https://jira.appcelerator.org/browse/TIMOB-24828) Unexpected layout for second UI element in a View

**TIMOB-24883**

```js
var win = Ti.UI.createWindow(),
    view0 = Ti.UI.createView({
        top: 20,
        bottom: 20,
        left: 20,
        right: 20,
        borderColor: 'blue',
        borderWidth: 1,
        backgroundColor: 'white'
    }),
    view = Ti.UI.createView({
        top: 10,
        bottom: 10,
        left: 10,
        right: 10,
        borderColor: 'green',
        borderWidth: 10,
        backgroundColor: 'red'
    });
view.add(view0);

win.add(view);
win.open();
```

timob-24883

**TIMOB-24848**

```js
var win = Ti.UI.createWindow({ backgroundColor: 'white' }),
    list = Ti.UI.createListView({
        templates: {
            template: {
                childTemplates: [
                    {
                        type: 'Ti.UI.Label',
                        bindId: 'title',
                        properties: {
                            left: '5dp',
                            right: '5dp'
                        }
                    }
                ]
            }
        },
        defaultItemTemplate: 'template'
    }),
    section = Ti.UI.createListSection(),
    items = [];

for (var i = 0; i < 3; i++) {
    items.push({
        title: { text: 'TEXT' }
    });
}
section.setItems(items);
list.setSections([section]);

win.add(list);
win.open();
```

Expected: This does not crash.

**TIMOB-24831**

```js
var _window = Ti.UI.createWindow();
var lbl = Ti.UI.createLabel({
    text:'number1. This is some long text. 2. This is some long text. 3. This is some long text. 4. This is some long text. 5. This is some long text. 6. This is some long text. 7. This is some long text. 8. This is some long text. 9. This is some long text. 10. This is some long text. 11. This is some long text. 12. This is some long text. 13. This is some long text. 14. This is some long text. 15. This is some long text. 16. This is some long text. 17. This is some long text. 18. This is some long text.',
    right: 10,
    left:10,
    top:10,
    bottom:10,
    borderWidth: 1,
    borderColor: 'blue'
});
_window.add(lbl);
_window.open();
```

Expected: Label should respect right, left, top and bottom properties.

**TIMOB-24828**

```js
var _window = Ti.UI.createWindow();
var txtUsername = Ti.UI.createTextField({
    top: 50,
    left: 10,
    right: 10,
    hintText: 'Username'
});
var txtPassword = Ti.UI.createTextField({
    top: 150,
    left: 10,
    right: 10,
    hintText: 'Password'
});
var view = Ti.UI.createView({
    height: 480,
    width: 320,
    borderColor: 'green',
    borderWidth: 5,
    backgroundColor: 'red'
});
view.add(txtPassword)
view.add(txtUsername);
_window.add(view);
_window.open();
```

Expected: `txtPassword` and `txtUsername` should have same width

timob-24828

### CODE REVIEW

File: Source/UI/src/WindowsViewLayoutDelegate.cpp [hunk 1]
```diff
@@ -1626,8 +1635,14 @@ namespace TitaniumWindows
 
 		void WindowsViewLayoutDelegate::onLayoutEngineCallback(Titanium::LayoutEngine::Rect rect, const std::string& name)
 		{
-			auto skipHeight = shouldUseOwnHeight() || (is_height_size__ && rect.height == 0);
-			auto skipWidth  = shouldUseOwnWidth()  || (is_width_size__  && rect.width  == 0);
+			if (parent__) {
+				const auto p_border = parent__->getViewLayoutDelegate()->get_borderWidth();
+				rect.x -= p_border;
```
Comment (garymathews) [inline Source/UI/src/WindowsViewLayoutDelegate.cpp]: `p_border` is an `std::string` and `rect.x` `rect.y` is a `double`

```
Error C2677 binary '-=': no global operator found which takes type 'const std::string' (or there is no acceptable conversion)
```

Comment (infosia) [inline Source/UI/src/WindowsViewLayoutDelegate.cpp]: Ah nice, updated. 👍

**Comment by ewanharris** [comment]
CI is green, merging

**Comment by ewanharris** [comment]
@infosia I've marked the tickets in this PR for **7.0.0**, but I think we really want them in **6.2.0** right? If so could you set up backports? I guess we'll have to do this for most PRs that are open sorry 🤕

**Comment by infosia** [comment]
@ewanharris grr right. I will push them later on.

**Pull request merged**